src/NodeApi/IJSValue.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#if !NET7_0_OR_GREATER
+using System;
+using System.Reflection;
+#endif
+
+namespace Microsoft.JavaScript.NodeApi;
+
+#if NET7_0_OR_GREATER
+// A static interface that helps with the conversion of JSValue to a specific type.
+public interface IJSValue<TSelf> where TSelf : struct, IJSValue<TSelf>
+{
+    public static abstract bool CanBeConvertedFrom(JSValue value);
+
+    public static abstract TSelf CreateUnchecked(JSValue value);
+}
+#else
+// A static class that helps with the conversion of JSValue to a specific type.
+public static class IJSValueShim<T> where T : struct
```

Comment:
This class can be `internal`. Or you could make the delegates private members of `JSValue`.

Reply:
Changed to be `internal` They cannot be part of `JSValue`. The delegates must be per `T`.